### PR TITLE
Add a run function to the base FastAPI for running the server via uvicorn from the base class.

### PR DESCRIPTION
**Pull request by Ixyk-Wolf**
Just added a run function to the base so it can run it using uvicorn without having the user need to import it and do it themself. Keeps all original arguments and raises an error if uvicorn isn't installed. Just wanted to do this because Flask has this and I have wanted it in FastAPI since I switched over.

### CODE REVIEW

File: fastapi/applications.py [hunk 1]
```diff
@@ -737,3 +737,10 @@ def trace(
             name=name,
             callbacks=callbacks,
         )
+    def run(self, **args):
+        try:
+            import uvicorn
+            uvicorn.run(self, **args)
+        except:
```
Comment (Kludex) [inline fastapi/applications.py:744]:
```suggestion
        except ImportError:
```

Comment (graingert) [reply fastapi/applications.py:744]:
```suggestion
        except ModuleNotFoundError:
```

**Comment by Ixyk-Wolf** [comment]
Sorry if I'm slow to respond in the next few weeks because I'll be really busy.

File: fastapi/applications.py [hunk 2]
```diff
@@ -769,6 +769,7 @@ def trace(
     def run(self, **args):
         try:
             import uvicorn
-            uvicorn.run(self, **args)
         except ImportError:
```
Comment (graingert) [inline fastapi/applications.py:772]:
```suggestion
        except ModuleNotFoundError:
```

Comment (graingert) [reply fastapi/applications.py:772]: Using this suggestion, this will raise a more informative diagnostic if uvicorn is installed, but fails to import

Comment (Ixyk-Wolf) [reply fastapi/applications.py:772]: Traceback (most recent call last):
  File "C:\Users\Isaac\stdinmedia\uvicorn.py", line 11, in <module>
    app.run()
  File "C:\Users\Isaac\stdinmedia\venv\lib\site-packages\fastapi\applications.py", line 771, in run
    import uvicorn
  File "C:\Users\Isaac\stdinmedia\uvicorn.py", line 11, in <module>
    app.run()
  File "C:\Users\Isaac\stdinmedia\venv\lib\site-packages\fastapi\applications.py", line 775, in run
    uvicorn.run(self, **args)
AttributeError: partially initialized module 'uvicorn' has no attribute 'run' (most likely due to a circular import)


This is the error that is raised when I name the project file uvicorn. It is neither an ImportError or a ModuleNotFoundError. I really don't see it.

Comment (graingert) [reply fastapi/applications.py:772]: Here's some more details behind the motivation of `ModuleNotFoundError`: https://bugs.python.org/issue15767

if `import uvicorn` raises an instance of `ImportError` that's not an instance of `ModuleNotFoundError` then installing uvicorn won't fix it

Comment (Ixyk-Wolf) [reply fastapi/applications.py:772]: I have just tested it, and see no difference. I'll just switch it I guess.

**Comment by Ixyk-Wolf** [comment]
I wonder if this should be a pull request for Scarlette so it is also in FastAPI. I think it would make more sense.

**Comment by vrslev** [comment]
I don’t think changes belong to a _ASGI framework_. `uvicorn` is one of many ASGI servers. And more importantly fine grained hierarchy would be broken if we’d couple ASGI servers and apps.

**Review by BilalAlpaslan — APPROVED** [review]

**Comment by tiangolo** [comment]
Thanks for the interest @Ixyk-Wolf! 🍰 

I prefer to teach how to use Uvicorn, and run it directly. If the added abstraction (in this case `app.run()`) and explaining how to use it doesn't simplify significantly using the lower level part (Uvicorn), I prefer to teach how to use the low level part than to create an abstraction on top.

Given that, I'll close this one, but thanks for the effort! ☕

**Pull request closed without merging**